lib/puppet/provider/firewalld_custom_service/firewall_cmd.rb

```diff
@@ -120,27 +118,21 @@ def protocols=(should)
     else
       to_remove = @property_hash[:protocols] - should
       ports_protos = []
-      unless @resource[:ports].include?(:unset)
-        ports_protos = Array(@resource[:ports]).select { |x| x['port'].nil? }.map { |x| x['protocol'] }
-      end
+      ports_protos = Array(@resource[:ports]).select { |x| x['port'].nil? }.map { |x| x['protocol'] } unless @resource[:ports].include?(:unset)
```

Comment:
I still think the guard cop here produces a too long line so I opened https://github.com/rubocop/rubocop/issues/10708